cognite/client/data_classes/_base.py

```diff
@@ -281,15 +278,33 @@ def get(self, id: Optional[int] = None, external_id: Optional[str] = None) -> Op
             return self._id_to_item.get(id)
         return self._external_id_to_item.get(external_id)
 
-    def to_pandas(self, camel_case: bool = False) -> pandas.DataFrame:
-        """Convert the instance into a pandas DataFrame.
+    def to_pandas(
+        self,
+        camel_case: bool = False,
+        expand_metadata: bool = False,
+        metadata_prefix: str = "metadata.",
+    ) -> pandas.DataFrame:
+        """Convert the instance into a pandas DataFrame. Note that if the metadata column is expanded and there are
+        keys in the metadata that already exist in the DataFrame, then an error will be raised by pd.join.
+
+        Args:
+            camel_case (bool): Convert column names to camel case (e.g. `externalId` instead of `external_id`)
+            expand_metadata (bool): Expand the metadata column into separate columns.
+            metadata_prefix (str): Prefix to use for metadata columns.
 
         Returns:
-            pandas.DataFrame: The dataframe.
+            pandas.DataFrame: The Cognite resource as a dataframe.
         """
         pd = cast(Any, utils._auxiliary.local_import("pandas"))
         df = pd.DataFrame(self.dump(camel_case=camel_case))
-        return convert_nullable_int_cols(df, camel_case)
+        df = convert_nullable_int_cols(df, camel_case)
+
+        if expand_metadata and "metadata" in df.columns:
+            # Equivalent to pd.json_normalize(df["metadata"]) but is a faster implementation.
+            meta_series = (meta := df.pop("metadata"))[meta.notna()]
```

Comment:
Last comment, not sure what I was thinking here lol
```suggestion
            meta_series = df.pop("metadata").dropna()
```

Reply:
...also not sure you need the comment regarding pd.json_normalize

Reply:
Well @doctrino mentioned it would be nice to add, and I thought it was fine to add too. 😄 🤷‍♂️ You are a wizard @haakonvt. 🧙‍♂️  So it is good to make sure people understand the where/why of the wizardry. In the (maybe distant) future, someone might replace this more syntactically-complex but algorithmically-more-efficient solution with a less-verbose but more-inefficient solution like `pd.json_normalize` without realising it is less efficient.